### PR TITLE
Microsoft.AspNetCore.SignalR.Redis API review

### PR DESCRIPTION
**Pull request by JamesNK**
- Rename RedisOptions.Factory to ConnectionFactory
- Rename RedisOptions.Options to Configuration
- Internalize Microsoft.AspNetCore.SignalR.Redis.Internal

### CODE REVIEW

File: benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks.csproj [hunk 1]
```diff
@@ -12,8 +12,11 @@
     <ProjectReference Include="..\..\src\Microsoft.AspNetCore.SignalR.Common\Microsoft.AspNetCore.SignalR.Common.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.AspNetCore.SignalR.Client.Core\Microsoft.AspNetCore.SignalR.Client.Core.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.AspNetCore.SignalR.Protocols.MessagePack\Microsoft.AspNetCore.SignalR.Protocols.MessagePack.csproj" />
-    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.SignalR.Redis\Microsoft.AspNetCore.SignalR.Redis.csproj" />
     <ProjectReference Include="..\..\test\Microsoft.AspNetCore.SignalR.Tests.Utils\Microsoft.AspNetCore.SignalR.Tests.Utils.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.SignalR.Redis\Microsoft.AspNetCore.SignalR.Redis.csproj">
+      <!-- Alias is required because internal MemoryBufferWriter is in multiple projects that InternalsVisibleTo this project -->
+      <Aliases>redis</Aliases>
```
Comment (davidfowl) [inline benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks.csproj]: Ughh wtf.

Comment (davidfowl) [reply benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks.csproj]: Maybe we should just pubternal instead?

Comment (JamesNK) [reply benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks.csproj]: So leave Microsoft.AspNetCore.SignalR.Redis.Internal as public?

btw with these shared internal files, and InternalVisibleTo, having some aliases will eventually have to happen unless something changes. MemoryBufferWriter is used everywhere

**Comment by davidfowl** [comment]
@JamesNK Can you move RedisLog to the Internal folder?

**Pull request merged**